### ISSUE TITLE
MetadataGenerator name collisions

### ISSUE THREAD

**Issue by PanayotCankov** [NativeScript/ios-runtime#233]
[From to-do in Meta/IdentifierFactory.cpp#L9](https://github.com/NativeScript/ios-metadata-generator/blob/0713b3eb8cbb3462a7d382d8188023db08220455/src/Meta/IdentifierFactory.cpp#L9)
[Reproduced with the KontaktSDK](http://cocoapods.org/pods/KontaktSDK)

In the metadata generator there are some static maps that contains hardcoded symbol names from the iOS SDK for which we know there is conflict in their JS name and must be renamed. Maybe there is better way to pass this map to the IdentifierGenerator. For third-party libraries the map may be parsed from file or something more flexible.

The most easy to consume way is if they can be generated from the decls, this won't involve manual setup. However it also involves some magic. Developers should be notified for such collisions and the way they are resolved. e.g. trace that: the TKTRegion interface collides with the TKTRegion protocol and the protocol is renamed to TKTRegionProtocol.

**Comment by ivanbuhov** [comment]
We have workaround for KontaktIO (0f67a2bb27c320e801fb7592bbcf39fe68729db0), but it is dirty solution. We just hardcode the names of the protocols which are in collision in our metadata generator. The solution is temporary until we find a better one.

**Comment by jasssonpet** [comment]
```objective-c
@protocol MyName
+ (void)method;
@end

@protocol MyNameProtocol
+ (void)method1;
@end

@interface MyName : NSObject
@end
```

In this case the MyName protocol can't seem to be visible in swift

**Comment by jasssonpet** [comment]
It seems that Swift has similar problems.

## Case 1

```objective-c
char *timezone(int, int);
extern long timezone;
```

```swift
print(timezone) // Ambigous use of 'timezone'`
```

## Case 2
```objective-c
@protocol MyName
+ (void)method;
@end

@protocol MyNameProtocol
+ (void)method1;
@end

@interface MyName : NSObject
@end
```

In this case the `MyName` protocol isn't visible in swift.

**Comment by fealebenpae** [comment]
Sounds fine to me.

**Comment by jasssonpet** [comment]
What do you think to report errors at runtime, instead of build time. Because we may sometimes fail the build, but if the user doesn't access this API, he wouldn't be able to do so.

**Comment by fealebenpae** [comment]
How would we know at runtime that there has been a name collision at build time?

**Comment by jasssonpet** [comment]
We could mark the name in the metadata that it is ambiguous or something.

**Comment by fealebenpae** [comment]
With a flag, perhaps? Alright, I think this will work.

**Comment by jasssonpet** [comment]
Those are some implementation details™. Lets clear the use case first :)

**Comment by fealebenpae** [comment]
Fixed in #299
